zerver/views/push_notifications.py

```diff
@@ -210,3 +216,12 @@ def self_hosting_auth_json_endpoint(
     redirect_url = self_hosting_auth_view_common(request, user_profile, next_page)
 
     return json_success(request, data={"billing_access_url": redirect_url})
+
+
+def self_hosting_auth_not_configured(request: HttpRequest) -> HttpResponse:
+    if settings.BILLING_ENABLED or uses_notification_bouncer():  # nocoverage
+        # This error page should only be available if the config error
+        # is actually real.
+        return render(request, "404.html", status=404)
+
+    return config_error(request, "remote_billing_bouncer_not_configured")
```

Comment:
It feels a bit hacky that this relies on return a json **success**, which happens to have a `billing_access_url` which will trick the desktop app into opening it in a new browser window, and thus showing a nice config error page. On the other hand, it's actually a pretty comfy way to solve the problem of how to make the desktop app handle it

Reply:
Yeah, I think that's probably fine.